content/inheritance/exercises/_index.md

```diff
@@ -0,0 +1,287 @@
+
+   The `Computer` class extends `Laptop` and `Smartphone`.  
+
+   Arrows should point towards `Computer` class.
+
+   Remember to add properties and methods to your diagram.
+
+{{% /expand %}}
+
+   
+1. For a **parent** class: add 3 fields, 2 methods, and a constructor.
+
+   {{% expand "Check your solution" %}}
+
+   The fields and methods selected for these examples can differ from your fields and methods.  
+
+   ```csharp
+      public class Computer
+      {
+         public double Ram { get; set; }
+         public double Storage { get; set; }
+         public readonly bool hasKeyboard;
+
+         public Computer(double ram, double storage, bool hasKeyboard)
+         {
+            Ram = ram;
+            Storage = storage;
+            this.hasKeyboard = hasKeyboard;
+         }
+
+         public double IncreaseRam(double extraRam)
+         {
+            return Ram += extraRam;
+         }
+
+         public double IncreaseStorage(double extraStorage)
+         {
+         return Storage += extraStorage;
+         }
+      }
+   ```
+
+   {{% /expand %}}
+
+
+
+1. For a **child** class: add at least 1 additional field and 1 additional method.
+
+   {{% expand "Check your solution" %}}
+
+   Here is a possible `Laptop` class.
+
+
+   ```csharp
+      public class Laptop : Computer
+      {
+         public double Weight { get; set; }
+
+         public Laptop(double ram, double storage, bool hasKeyboard, double weight) : base(ram, storage, hasKeyboard)
+         {
+         Weight = weight;
+         }
+
+         public bool IsClunky()
+         {
+            if (Weight > 5.0)
+            {
+               return true;
+            }
+            else
+            {
+               return false;
+            }
+         }
+      }
+   ```
+   {{% /expand %}}
+
+
+   {{% expand "Check your solution" %}}
+
+   Here is a possible `Smartphone` class.
+
+   ```csharp
+      
+      public class SmartPhone : Computer
+      {
+         public int NumberOfSelfies { get; set; }
+         public SmartPhone(int ram, int storage, bool hasKeyboard, int numberOfSelfies) : base(ram, storage, hasKeyboard)
+         {
+            NumberOfSelfies = numberOfSelfies;
+         }
+         
+         public void TakeSelfie()
+         {
+            NumberOfSelfies++;
+         }
+      }
+   ```
+   {{% /expand %}}
+
+
+1. At least one of your fields should be either `static`, `readonly`, or a `const`.
+
+   {{% expand "Check your solution" %}}
+
+   ```csharp{linenos=table,hl_lines=[5],linenostart=1} 
+      public class Computer
+      {
+         public double Ram { get; set; }
+         public double Storage { get; set; }
+         public readonly bool hasKeyboard;
+   
+      // code continues
+      
+   ```
+   {{% /expand %}}
+
+## Class implementation
+Implement your design and add unit tests to a new TechnologyTests MSTest project.
+
+Add a new MSTest project to your solution.
+{{% expand "Check your solution" %}}
```

Comment:
Please tab these in two match the location of the ordered list Check your solution.

Reply:
done.